cns/cni/initialize.go

```diff
@@ -0,0 +1,36 @@
+package cni
+
+import (
+	"github.com/Azure/azure-container-networking/cni/api"
+	"github.com/Azure/azure-container-networking/cni/client"
+	"github.com/Azure/azure-container-networking/cns"
+	"k8s.io/utils/exec"
+)
+
+// CNICNSInitializer returns an implementation of cns.PodInfoByIPProvider
+// that execs out to the CNI and uses the response to build the PodInfo map.
+func CNICNSInitializer() (cns.PodInfoByIPProvider, error) {
+	cli := client.New(exec.New())
+	state, err := cli.GetEndpointState()
```

Comment:
@matmerr, does CNI acquires the lock in GetEndpointState?

Reply:
it looks like it does since CNI is just invoked normally and it always takes the lock